src/Analysis/Codelyzer.Analysis/CodeAnalyzer.cs

```diff
@@ -41,6 +41,13 @@ protected CodeAnalyzer(AnalyzerConfiguration configuration, ILogger logger)
         /// <returns></returns>
         public abstract Task<List<AnalyzerResult>> AnalyzeSolution(string solutionPath);
 
+        /// <summary>
+        /// Runs analysis on a solution
+        /// </summary>
+        /// <param name="solutionPath">The path to the solution file</param>
+        /// <returns></returns>        
+        public abstract IAsyncEnumerable<AnalyzerResult> AnalyzeSolutionGenerator(string solutionPath);
+        
```

Comment:
Similar to above: suggested rename to "AnalyzeSolutionWithGeneratorAsync"

Reply:
> General question: we have multiple ways to analyze a project now and they seem to share some code- is there a way to put common functionality in a shared method and would it make sense?
> 
> The generator functionality is a great enhancement otherwise 👍

Some of these might make sense, but the latest addition (generator) will have to be separate. The logic for this might expand to the build process (so we are building separately) so it'll diverge from our current code.

As for the other classes, we have an open issue to refactor some of this code. Please feel free to create other issues or PRs to address.

Reply:
Got it, that makes sense